src/cargo/ops/cargo_check.rs

```diff
@@ -0,0 +1,13 @@
+use core::Workspace;
+use util::important_paths::find_root_manifest_for_wd;
+use util::{CliResult, Config};
+
+pub fn with_check_ws<F>(flag_manifest_path: Option<String>,
+                        config: &Config, f: F)
+                        -> CliResult<Option<()>>
+    where F: FnOnce(&Workspace) -> CliResult<Option<()>>
+{
+    let root = find_root_manifest_for_wd(flag_manifest_path, config.cwd())?;
+    let ws = Workspace::new(&root, config)?;
```

Comment:
This seems like not a lot of code to inline locally perhaps? If we keep this function, though, could it just return a `Workspace` instead of taking a closure?